### ISSUE TITLE
Models in config not being used

### ISSUE THREAD

**Issue by LuBolin** [EpistasisLab/tpot#979]
I am trying to track score of each model every so often during a run, so I am doing little generations with warm start. I am having problems with TPOTRegressor where RidgeCV is not being used when fitting.

## Context of issue

I am trying to use warm_start to track score of multiple regressor models every few generations. However, even though it tells me that 3 operators have been imported by TPOT, I do not see RidgeCV in the evaluated_individuals.
Sorry if my issue is badly formatted or something, this is my first legitimate GitHub issue.

Thank you.

Here is my config:
test_config = {
    "sklearn.linear_model.RidgeCV": {
        "alphas": [0.1, 1.0, 10.0],
        "fit_intercept": [True, False],
        "normalize":[True,False],
        "scoring":["neg_mean_squared_error","r2"],
        "store_cv_values":[True,False]
    },
    "sklearn.ensemble.RandomForestRegressor": {
        "n_estimators": np.arange(10, 201, 5),
        "max_features": np.arange(0.05, 1.01, 0.05),
        "min_samples_split": range(2, 21),
        "min_samples_leaf": range(1, 21),
        "bootstrap": [True, False],
    },
    "xgboost.XGBRegressor": {
        "objective": ['reg:squarederror'],
        "n_estimators": np.arange(10, 201, 5),
        "max_depth": range(1, 21),
        "learning_rate": [1e-3, 1e-2, 1e-1, 0.5, 1.0],
        "subsample": np.arange(0.05, 1.01, 0.05),
        "min_child_weight": range(1, 21),
        "nthread": [1],
    },
}

Here is my generated dummy data:
independent=np.random.randint(100,size=1000)
regression_dependent=np.random.randint(1000,size=1000)
X_train2, X_test2, Y_train2, Y_test2 = train_test_split(
    independent, regression_dependent, train_size=0.7, test_size=0.3
)

Here is my regressor:
testR = TPOTRegressor(
    generations=3,
    population_size=10,
    verbosity=3,
    config_dict=test_config,
    scoring="neg_mean_squared_error",
    random_state=123,
    use_dask=True,
    template="Regressor",
    warm_start=True
)

And here is my function to read the evaluated individuals:
for i in range(5):
    testR.fit(X_train2.reshape(-1,1), Y_train2)
    tuples=list(testR.evaluated_individuals_.items())
    tuples.sort(key=lambda x: x[1]["internal_cv_score"], reverse=True)
    shownModels = []
    for x in tuples:
        pipeline = x[0]
        name = pipeline[: pipeline.find("(")]
        if name in shownModels:
            continue
        shownModels.append(name)
    models=','.join(str(e) for e in shownModels)
    print(models," are shown in run ",i)

## Process to reproduce the issue
1.Create a config with RidgeCV and some settings.
2.Create a TPOTRegressor using the just created config, on verbosity=3
3.Fit on regression data

## Expected result
Under verbosity=3, see all(3) models from config being ran.
See all(3) unique models in evaluated_individuals_

## Current result
Under verbosity=3, only see RandomForestRegressor and XGBRegressor runs.
evaluated_individuals_ only contain randomforestregressor and xgbregressor.

To make output cleaner, I am pasting results I get with verbosity 0 and 3 separately.

Verbosity=0:
XGBRegressor,RandomForestRegressor  are shown in run  0
XGBRegressor,RandomForestRegressor  are shown in run  1
XGBRegressor,RandomForestRegressor  are shown in run  2
XGBRegressor,RandomForestRegressor  are shown in run  3
XGBRegressor,RandomForestRegressor  are shown in run  4

Verbosity=3:
3 operators have been imported by TPOT.

HBox(children=(IntProgress(value=0, description='Optimization Progress', max=40, style=ProgressStyle(descripti…

Generation 1 - Current Pareto front scores:
-1	-89913.48782333538	RandomForestRegressor(input_matrix, RandomForestRegressor__bootstrap=True, RandomForestRegressor__max_features=0.25, RandomForestRegressor__min_samples_leaf=18, RandomForestRegressor__min_samples_split=5, RandomForestRegressor__n_estimators=20)

Generation 2 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Generation 3 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

XGBRegressor,RandomForestRegressor  are shown in run  0
3 operators have been imported by TPOT.

HBox(children=(IntProgress(value=0, description='Optimization Progress', max=40, style=ProgressStyle(descripti…

Generation 1 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Generation 2 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Generation 3 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

XGBRegressor,RandomForestRegressor  are shown in run  1
3 operators have been imported by TPOT.

HBox(children=(IntProgress(value=0, description='Optimization Progress', max=40, style=ProgressStyle(descripti…

Generation 1 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Generation 2 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Generation 3 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

XGBRegressor,RandomForestRegressor  are shown in run  2
3 operators have been imported by TPOT.

HBox(children=(IntProgress(value=0, description='Optimization Progress', max=40, style=ProgressStyle(descripti…

Generation 1 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Generation 2 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Generation 3 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

XGBRegressor,RandomForestRegressor  are shown in run  3
3 operators have been imported by TPOT.

HBox(children=(IntProgress(value=0, description='Optimization Progress', max=40, style=ProgressStyle(descripti…

Generation 1 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Generation 2 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
_pre_test decorator: _random_mutation_operator: num_test=0 len() of unsized object.
Pipeline encountered that has previously been evaluated during the optimization process. Using the score from the previous evaluation.
Generation 3 - Current Pareto front scores:
-1	-88616.43101169747	XGBRegressor(input_matrix, XGBRegressor__learning_rate=0.1, XGBRegressor__max_depth=3, XGBRegressor__min_child_weight=7, XGBRegressor__n_estimators=30, XGBRegressor__nthread=1, XGBRegressor__objective=reg:squarederror, XGBRegressor__subsample=1.0)

XGBRegressor,RandomForestRegressor  are shown in run  4

**Comment by weixuanfu** [comment]
I have a quick look of the codes. There should be a bug in the configuration of `RidgeCV`: it should be `"alphas": [(0.1, 1.0, 10.0)],` instead of `"alphas": [0.1, 1.0, 10.0],`.

Please try the demo below to check if it works in your environment. 
```python
from tpot import TPOTRegressor
from sklearn.datasets import load_boston
from sklearn.model_selection import train_test_split

housing = load_boston()
X_train, X_test, y_train, y_test = train_test_split(housing.data, housing.target,
                                                    train_size=0.75, test_size=0.25, random_state=42)
test_config = {
    "sklearn.linear_model.RidgeCV": {
        "alphas": [(0.1, 1.0, 10.0)],
        "fit_intercept": [True, False],
        "normalize":[True,False],
        "scoring":["neg_mean_squared_error","r2"],
        "store_cv_values":[True,False]
    }
}

tpot = TPOTRegressor(generations=5, 
                    population_size=50, 
                    verbosity=2, 
                    random_state=42,
                    config_dict=test_config)
tpot.fit(X_train, y_train)
print(tpot.score(X_test, y_test))

```

**Comment by LuBolin** [comment]
Yes it worked after adding the brackets. I do not understand why though.
Looking at other models, the values for variables to be tested are directly in a list or numpy array, instead of being a tuple within a list which has only 1 element.
May I know why is it that RidgeCV requires such a setting? Thank you.

**Comment by weixuanfu** [comment]
Please check the alphas param in [RidgeCV](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.RidgeCV.html), which should a ndarray of shape (n_alphas,) and default alphas is `(0.1, 1.0, 10.0)` (a tuple), so the configuration of `alphas` should be a list of tuples, e.g. `[(0.1, 1.0, 10.0), (0.2, 2.0, 5.0)]`

**Comment by LuBolin** [comment]
I see, my bad for not reading documentation carefully. Thank you so much.
